sqlmesh/core/engine_adapter/base.py

```diff
@@ -1187,23 +1246,131 @@ def scd_type_2(
             columns_to_types = self.columns(target_table)
         if not columns_to_types:
             raise SQLMeshError(f"Could not get columns_to_types. Does {target_table} exist?")
-        if updated_at_name not in columns_to_types:
+        if not unique_key:
+            raise SQLMeshError("unique_key must be provided for SCD Type 2")
+        if check_columns and updated_at_name:
+            raise SQLMeshError(
+                "Cannot use both `check_columns` and `updated_at_name` for SCD Type 2"
+            )
+        if check_columns and updated_at_as_valid_from:
+            raise SQLMeshError(
+                "Cannot use both `check_columns` and `updated_at_as_valid_from` for SCD Type 2"
+            )
+        if execution_time_as_valid_from and not check_columns:
+            raise SQLMeshError(
+                "Cannot use `execution_time_as_valid_from` without `check_columns` for SCD Type 2"
+            )
+        if updated_at_name and updated_at_name not in columns_to_types:
             raise SQLMeshError(
                 f"Column {updated_at_name} not found in {target_table}. Table must contain an `updated_at` timestamp for SCD Type 2"
             )
+
         unmanaged_columns = [
             col for col in columns_to_types if col not in {valid_from_name, valid_to_name}
         ]
         time_data_type = columns_to_types[valid_from_name]
         select_source_columns: t.List[t.Union[str, exp.Alias]] = [
             col for col in unmanaged_columns if col != updated_at_name
         ]
-        select_source_columns.append(exp.cast(updated_at_name, time_data_type).as_(updated_at_name))
-        valid_from_start = (
-            updated_at_name
-            if updated_at_as_valid_from
-            else self._to_utc_timestamp("1970-01-01 00:00:00+00:00", time_data_type)
-        )
+        if updated_at_name:
+            select_source_columns.append(
+                exp.cast(updated_at_name, time_data_type).as_(updated_at_name)
+            )
+
+        # If a star is provided, we include all unmanaged columns in the check.
+        # This unnecessarily includes unique key columns but since they are used in the join, and therefore we know
+        # they are equal or not, the extra check is not a problem and we gain simplified logic here.
+        # If we want to change this, then we just need to check the expressions in unique_key and pull out the
+        # column names and then remove them from the unmanaged_columns
+        if check_columns and check_columns[0] == exp.Star():
+            check_columns = [exp.to_column(col) for col in unmanaged_columns]
```

Comment:
this to_column conversion isn't necessary

Reply:
`unmanaged_columns` is a list of strings and `check_columns` is a list of expression (or columns after update) so it does need this.

Reply:
change the type of check_columns to exp.Column

Reply:
That is what I said here: "is a list of expression (or columns after update)"

So we are agreeing this is needed right?

Reply:
the call exp.column(col) instead

Reply:
if it's a string (unmanaged_columns), then you should use exp.column, but if it's a column, you don't need it and should just fix the type checking

Reply:
Yeah so it is a string of columns. So it sounds like `exp.to_column` should change to `exp.column`